locks/dynamodb/dynamo_lock_table_test.go

```diff
@@ -106,6 +107,22 @@ func TestWriteAndRemoveItemFromLockTable(t *testing.T) {
 
 		// Finally, check the item no longer exists
 		assertItemNotExistsInTable(t, itemId, tableName, client)
+
+		// Additionally, verify removing an item that doesn't exist returns an expected error
+		err = removeItemFromLockTable(itemId, tableName, client)
+		if assert.NotNil(t, err, "Error expected") {
+			awsErr, isAwsErr := errors.Unwrap(err).(awserr.Error)
+			assert.True(t, isAwsErr, "Expected error to be an awserr error")
+			assert.Equal(t, awsErr.Code(), "ConditionalCheckFailedException", "Unexpected awsErr code")
+		}
+
+		// And then, verify removing an item from a non-existent table returns an expected error
+		err = removeItemFromLockTable(itemId, uniqueTableNameForTest(), client)
+		if assert.NotNil(t, err, "Error expected") {
+			awsErr, isAwsErr := errors.Unwrap(err).(awserr.Error)
+			assert.True(t, isAwsErr, "Expected error to be an awserr error")
+			assert.Equal(t, awsErr.Code(), "ResourceNotFoundException", "Unexpected awsErr code")
+		}
```

Comment:
If you update the `removeItemFromLockTable` to return the custom error type as mentioned above, then this test code could verify that it gets those clear error messages back rather than parsing the AWS errors here a second time.